pkg/controller/windowsmachineconfig/nodeconfig/nodeconfig.go

```diff
@@ -55,10 +56,21 @@ func discoverKubeAPIServerEndpoint() (string, error) {
 	if err != nil {
 		return "", errors.Wrap(err, "unable to get config to talk to kubernetes api server")
 	}
-	if len(cfg.Host) == 0 {
+
+	client, err := clientset.NewForConfig(cfg)
+	if err != nil {
+		return "", errors.Wrap(err, "unable to get client from the given config")
+	}
+
+	host, err := client.ConfigV1().Infrastructures().Get("cluster", metav1.GetOptions{})
+	if err != nil {
+		return "", errors.Wrap(err, "unable to get cluster infrastructure resource")
+	}
+
+	if len(host.Status.APIServerURL) == 0 {
```

Comment:
I don't understand why this is necessary. Please make this a separate commit explaining the necessity in the commit message. This statement includes all of the role changes as well.

How is this working in CI? Is the operator pod using the mounted kubeconfig?

Reply:
Okay, I see now. The KUBECONFIG env variable is being set and that takes higher priority than the pod permissions. 

I understand why now. The commit split is still necessary though :)

Reply:
This is required because, when we use pod permission, the config object returns host in the format `172.X.X.X` instead of `api server url`. Using `infrastructure` resource we can access  `APIServerURL`.

Reply:
Should we use `APIServerInternalURL` or `APIServerURL`

Reply:
I will change this to `APIServerInternalURL`